R/forecast.R

```diff
@@ -80,3 +115,42 @@ run_and_forecast <- function(model_params, filter, index, n_particles,
   }
   trajectories
 }
+
+##' @export
+plot.mcstate_forecast <- function(x, ..., what = "1", data = NULL,
+                                  ylab = NULL, title = NULL, col = "grey80") {
+  partition_index <- as.integer(what)
```

Comment:
Larger and more potentially controversial change - in order to make the plotting testable, can you factor this into all the bits that *would* be passed into the underlying plotting functions, then have another small basically logic-free function for doing the plotting. Created data here might look like:

```
list(particles, title, xlab, ylab, data)
```

(as a named list of course)

Reply:
(The alternative is to get into plot diffing...)

Reply:
Leaving for now, raised #15 to track